clippy_lints/src/matches/mod.rs

```diff
@@ -974,12 +974,16 @@ impl_lint_pass!(Matches => [
 
 impl<'tcx> LateLintPass<'tcx> for Matches {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) {
-        if in_external_macro(cx.sess(), expr.span) {
+        if is_expn_of(expr.span, "matches").is_none() && in_external_macro(cx.sess(), expr.span) {
```

Comment:
I'm not sure if this should be `is_direct_expn_of` (also below) Otherwise it might also lint, if `matches!` is used in an external macro.

Reply:
Thanks for your response, I'll try this and push the change